client/findproviders.go

```diff
@@ -56,23 +56,30 @@ func (fp *Client) FindProvidersAsync(ctx context.Context, key cid.Cid) (<-chan F
 	if err != nil {
 		return nil, err
 	}
+
 	parsedRespCh := make(chan FindProvidersAsyncResult, 1)
 	go func() {
 		defer close(parsedRespCh)
-		if ctx.Err() != nil {
-			return
-		}
-		protoAsyncResp, ok := <-protoRespCh
-		if !ok {
-			return
-		}
+
+		// TODO wrap in a for loop after fixing https://github.com/ipld/go-ipld-prime/issues/374
```

Comment:
I think we should wrap the for loop now, even though multiple responses are not expected until the fix. This will keep this implementation correct and things will just work when the IPLD issue is fixed. Also, since there is no backlink to this comment in the IPLD issue, we risk forgetting to do this when the IPLD issue is addressed.